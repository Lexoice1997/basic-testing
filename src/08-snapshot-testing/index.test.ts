import { generateLinkedList } from './index';

describe('generateLinkedList', () => {
  interface User {
    id: number;
    name: string;
  }

  const firstClass: User[] = [
    { id: 1, name: 'Azamat' },
    { id: 2, name: 'Dawlet' },
  ];
  const secondClass: User[] = [
    { id: 3, name: 'Begis' },
    { id: 4, name: 'Islam' },
  ];

  const expectedLinkedList = {
    value: { id: 1, name: 'Azamat' },
    next: {
      value: { id: 2, name: 'Dawlet' },
      next: {
        value: null,
        next: null,
      },
    },
  };
  test('should generate linked list from values 1', () => {
    expect(generateLinkedList(firstClass)).toStrictEqual(expectedLinkedList);
  });

  test('should generate linked list from values 2', () => {
    expect(generateLinkedList(secondClass)).toMatchSnapshot();
  });
});
