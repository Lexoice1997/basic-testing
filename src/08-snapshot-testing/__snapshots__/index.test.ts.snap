// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateLinkedList should generate linked list from values 2 1`] = `
{
  "next": {
    "next": {
      "next": null,
      "value": null,
    },
    "value": {
      "id": 4,
      "name": "Islam",
    },
  },
  "value": {
    "id": 3,
    "name": "Begis",
  },
}
`;
